### PR TITLE
Feature/pep612 support

### PR DESCRIPTION
**Pull request by Filienko**
Modifying the current package setup to use pyproject.toml with accordance to [PEP621](https://peps.python.org/pep-0621/) standard and adding required example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,9 +30,26 @@ The Migration module provides methods to manage database migrations in a structu
...
+
+# Register the migration blueprint with your Flask app
+app.register_blueprint(migration_blueprint)
+</pre>
+
 ## Usage
 
-To initialize the Migration service and use a specific directory for migration scripts, you need to provide the path to the directory when creating an instance of the Migration class. By default, the service looks for migration scripts in the versions directory located in the same directory as the script.
+To initialize the Migration service and use a specific directory for migration scripts, you need to provide the path to the directory when creating an instance of the Migration class. By default, the service looks for scripts in examples directory, and can be altered via specifying MIGRATION_SCRIPTS_DIR enviroment var.  
+- For writing new migrations: the random UUID for the migration's revision id is generated automatically and does not need to be specified by user. The upgrade method is run on `upgrade` flask command, downgrade method is run on `downgrade` flask command. The method declaration for both is provided, but method functionality needs to be filled by the user.
```
Comment (pbugni) [inline README.md]: unless i'm overlooking it, you don't mention `how` to generate the migration script here?

File: README.md [hunk 2]
```diff
@@ -30,9 +30,26 @@ The Migration module provides methods to manage database migrations in a structu
...
+app = Flask(__name__)
+
+# Register the migration blueprint with your Flask app
+app.register_blueprint(migration_blueprint)
+</pre>
+
 ## Usage
 
-To initialize the Migration service and use a specific directory for migration scripts, you need to provide the path to the directory when creating an instance of the Migration class. By default, the service looks for migration scripts in the versions directory located in the same directory as the script.
+To initialize the Migration service and use a specific directory for migration scripts, you need to provide the path to the directory when creating an instance of the Migration class. By default, the service looks for scripts in examples directory, and can be altered via specifying MIGRATION_SCRIPTS_DIR enviroment var.
```
Comment (pbugni) [inline README.md]: s/in examples/in the nested examples/

File: README.md (hunk 1 above)
Comment (pbugni) [reply README.md]: something like "see the `migration` command, below

File: README.md [hunk 3]
```diff
@@ -54,21 +71,65 @@ migration_service.run_migrations(direction="downgrade")
...
+│   └── utils.py  
+├── examples/
+│   ├── add_mrn.py
+│   ├── add_identifier.py
+│   └── create_patient.py 
+```
 ## Example
 
-Provided is a simple migration script that creates a test patient and then modifies it. The operation was limited to one patient to limit potential interference with existing stores. However, the commands used in the examples can be expanded to any number of examples and were limited for the sake of demonstration.
+Provided is a simple set of migration scripts that create a patient and then modify it. The operation was limited to one patient to limit potential interference with existing stores. However, the commands used in the examples can be expanded to any number of examples and were limited for the sake of demonstration.
```
Comment (pbugni) [inline README.md]: s/set of/set of example/

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+[build-system]
+requires = ["setuptools>=46.1.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "fhir_migrations"
+version = "0.0.1"
+
+description = "The module captures essential methods for system migration, including running downgrade functions for stepping back during downgrades and applying upgrades, which execute upgrade functions for all unapplied migrations, while ensuring that existing migrations are applied before creating new ones."
```
Comment (pbugni) [inline pyproject.toml]: I'd expand your description to include details about being designed to work with a FHIR store.  How it tracks the latest upgrade run against your store in a Basic FHIR resource, etc.

**Review by pbugni — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -54,21 +72,65 @@ migration_service.run_migrations(direction="downgrade")
...
-├── migrations/
-│   ├── migration_resource.py
-│   ├── migration_utils.py
-│   └── versions/
-│       ├── 0001_initial_migration.py
-│       └── 0002_add_field_to_table.py
+On Unix-based systems, you can set the `MIGRATION_SCRIPTS_DIR` environment variable like this:
+   `export MIGRATION_SCRIPTS_DIR="/path/to/custom/migration_scripts_directory"`
+Or in the corresponding Dockerfile of your service by specifying MIGRATION_SCRIPTS_DIR 
+   `ENV MIGRATION_SCRIPTS_DIR=$SCRIPT_DIR`
```
Comment (ivan-c) [inline README.md]: I **think** you intended these to be separate lines, which needs slightly different markdown

```suggestion
On Unix-based systems, you can set the `MIGRATION_SCRIPTS_DIR` environment variable like this:

    export MIGRATION_SCRIPTS_DIR="/path/to/custom/migration_scripts_directory"

Or in the corresponding Dockerfile of your service by specifying MIGRATION_SCRIPTS_DIR 

    ENV MIGRATION_SCRIPTS_DIR=$SCRIPT_DIR
```

File: fhir_migrations/examples/add_mrn.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+def downgrade():
+    for record in patient_mrn_map:
+        pat_id = record['PAT_ID']
+        mrn = record['MRN']
+        remove_mrn_from_patient(pat_id, mrn)
+
+def remove_mrn_from_patient(pat_id, mrn):
+    # Fetch the existing patient resource
+    response = requests.get(f'{FHIR_SERVER_URL}/Patient?identifier=uwDAL_Clarity|{pat_id}', headers=HEADERS)
+    if response.status_code == 200:
```
Comment (ivan-c) [inline fhir_migrations/examples/add_mrn.py]: you might try a [guard clause](https://en.wikipedia.org/wiki/Guard_(computer_science)) here so the code is flatter. ie inverting the condition to fail on non-200 earlier

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,61 @@
+[build-system]
+requires = ["setuptools>=46.1.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "fhir_migrations"
+version = "0.0.1"
```
Comment (ivan-c) [inline pyproject.toml]: we typically try to manage the versioning via git/VCS. using a static version here may conflict with the VCS reports
```suggestion
dynamic = ["version"]
```

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+- tracks the latest upgrade run against your store in a Basic FHIR resource
+- updates the tracking resource with every new migration run
+
+The module can be used both to generate basic migration code,
+which then should be updated by the user, and to track the migration process.
+'''
+
+# TODO: ask for name
+authors = [
+    { name="CIRG", email="cirg@example.com" }
```
Comment (ivan-c) [inline pyproject.toml]:
```suggestion
    { name="CIRG", email="cirg-adm@uw.edu" }
```

**Review by ivan-c — CHANGES_REQUESTED** [review]
looks good! the version may be a blocker, but all other comments are nitpicks.

@Filienko please add a description to the PR when you get a chance

thank you!

**Review by ivan-c — APPROVED** [review]
looks great, thanks again!!

**Pull request merged**